src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -81,6 +81,17 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Sorts the internalList before casting as immutable.
+     * Returns an unmodifiable java List view with elements cast as immutable
+     * For use with other methods/libraries.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     */
+    public List<ReadOnlyPerson> sortedListView() {
+        internalList.sort((p1, p2) -> p1.getName().toString().compareToIgnoreCase(p2.getName().toString()));
```

Comment:
You should find a way to fully utilize the already given Name class's attribute fullName, instead of just using toString() so as to improve readability slightly.